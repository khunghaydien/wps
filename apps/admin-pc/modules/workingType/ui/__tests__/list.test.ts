import snapshotDiff from 'snapshot-diff';

import reducer, {
  actions,
  INITIALIZE,
  initialState,
  RESET_SELECTED_CODE,
  SET_SELECTED_CODE,
  State,
} from '../list';

describe('reducer()', () => {
  test('@@INIT', () => {
    // @ts-ignore
    const next = reducer(undefined, { type: '@@INIT' });
    expect(snapshotDiff({}, next)).toMatchSnapshot();
  });
  test(INITIALIZE, () => {
    // Arrange
    const prev: State = {
      selectedCode: 'code-0',
    };
    // Execute
    const next = reducer(prev, actions.initialize());
    // Assert
    expect(snapshotDiff(prev, next)).toMatchSnapshot();
  });
  test(SET_SELECTED_CODE, () => {
    // Arrange
    const prev = initialState;
    // Execute
    const next = reducer(prev, actions.setSelectedIndex('code-1'));
    // Assert
    expect(snapshotDiff(prev, next)).toMatchSnapshot();
  });
  test(RESET_SELECTED_CODE, () => {
    // Arrange
    const prev = {
      selectedCode: 'code-0',
    };
    // Execute
    const next = reducer(prev, actions.resetSelectedIndex());
    // Assert
    expect(snapshotDiff(prev, next)).toMatchSnapshot();
  });
});
