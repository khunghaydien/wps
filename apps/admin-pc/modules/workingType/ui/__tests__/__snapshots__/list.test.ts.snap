// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer() @@INIT 1`] = `
"Snapshot Diff:
- First value
+ Second value

- Object {}
+ Object {
+   \\"selectedCode\\": \\"\\",
+ }"
`;

exports[`reducer() ADMIN-PC/MODULES/WORKING_TYPE/UI/LIST/INITIALIZE 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"selectedCode\\": \\"code-0\\",
+   \\"selectedCode\\": \\"\\",
  }"
`;

exports[`reducer() ADMIN-PC/MODULES/WORKING_TYPE/UI/LIST/RESET_SELECTED_CODE 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"selectedCode\\": \\"code-0\\",
+   \\"selectedCode\\": \\"\\",
  }"
`;

exports[`reducer() ADMIN-PC/MODULES/WORKING_TYPE/UI/LIST/SET_SELECTED_CODE 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"selectedCode\\": \\"\\",
+   \\"selectedCode\\": \\"code-1\\",
  }"
`;
